assign-doi-datacite/src/test/java/no/unit/nva/doi/datacite/mdsclient/DataCiteMdsConnectionTest.java

```diff
@@ -94,6 +93,17 @@ public void postDoiSuccessfullyReturnsBodyInResponse()
         assertResponseContainsBody(httpResponse);
     }
 
+    @Test
+    public void postDoiWithoutBodyThrowsNullPointerException()
+        throws IOException, InterruptedException {
+        String body = IoUtils.stringFromResources(Path.of(DATACITE_MDS_OK_RESPONSE));
+        stubHttpClientWithHttpResponse(null);
+
+        DataCiteMdsConnection mockDataCiteMdsConnection = createDataCiteMdsConnection();
+
+        assertThrows(NullPointerException.class, () -> mockDataCiteMdsConnection.postMetadata(MOCK_DOI, NO_METADATA));
```

Comment:
The exception does not contain what was empty. It would be better if we included a message to say what is null. `Objects.requireNonNull` accepts a message as well, so we should better include one.

Reply:
Fixed